editor/filesystem_dock.cpp

```diff
@@ -3935,9 +3896,9 @@ void FileSystemDock::_project_settings_changed() {
 }
 
 void FileSystemDock::set_file_sort(FileSortOption p_file_sort) {
-	for (int i = 0; i != FILE_SORT_MAX; i++) {
-		tree_button_sort->get_popup()->set_item_checked(i, (i == (int)p_file_sort));
-		file_list_button_sort->get_popup()->set_item_checked(i, (i == (int)p_file_sort));
+	for (int i = 0; i != static_cast<int>(FileSortOption::FILE_SORT_MAX); i++) {
+		tree_button_sort->get_popup()->set_item_checked(i, (i == static_cast<int>(p_file_sort)));
+		file_list_button_sort->get_popup()->set_item_checked(i, (i == static_cast<int>(p_file_sort)));
```

Comment:
I changed the enum to an enum class and that required the cast

Reply:
Then I'd question whether that was a good suggestion, CC @Geometror. 
If the code is to be made less readable then we need to have clear upsides for what changing to `enum class` provides.

Reply:
Does `int(value)` work here?

Reply:
I'd say it belongs inside `FileInfo` *or* as `enum class` to not pollute namespaces

Reply:
I did this changes because I got compilation errors. And then I followed the recommendation from [here](https://en.cppreference.com/w/cpp/language/enum)
> Each enumerator becomes a named constant of the enumeration's type (that is, name), which is contained within the scope of the enumeration, and can be accessed using scope resolution operator. There are no implicit conversions from the values of a scoped enumerator to integral types, although [static_cast](https://en.cppreference.com/w/cpp/language/static_cast) may be used to obtain the numeric value of the enumerator.

Reply:
But any cast should do. cpp reference will recommend `static_cast<>`, because it's more idiomatic for C++.

Reply:
Changed to cast using `(int)`

Reply:
While it requires explicit conversions I'd argue that in any case `enum class` should be used due to type safety and avoiding namespace collisions.